File: src/dataset/typescript/difference-of-squares_Exercism-typescript-thewanionly/gemini/GeminiFlash_difference-of-squares_v3.ts
export class Squares {
  private readonly count: number;
  private readonly sum: number;
  private readonly sumOfSquaresValue: number;

  constructor(count: number) {
    this.count = count;
    this.sum = (count * (count + 1)) / 2;
    this.sumOfSquaresValue = (count * (count + 1) * (2 * count + 1)) / 6;
  }

  get sumOfSquares(): number {
    return this.sumOfSquaresValue;
  }

  get squareOfSum(): number {
    return this.sum * this.sum;
  }

  get difference(): number {
    return this.squareOfSum - this.sumOfSquares;
  }
}

File: src/dataset/c/difference_of_squares_Exercism_ThomasZumsteg/gemini/GeminiFlash_difference_of_squares_v3.c
#include "difference_of_squares.h"

int sum_of_squares(int n) {
    return n * (n + 1) * (2 * n + 1) / 6;
}

int square_of_sum(int n) {
    int sum = n * (n + 1) / 2;
    return sum * sum;
}

int difference_of_squares(int n) {
    return n * (n + 1) * (n * n - n - 1) / 12;
}